src/client_side_request.cc

```diff
@@ -530,6 +536,7 @@ ClientRequestContext::hostHeaderVerify()
         debugs(85, 3, "FAIL on validate URL domain length " << http->request->url.host() << " matches Host: " << host);
         hostHeaderVerifyFailed(host, http->request->url.host());
     } else if (matchDomainName(host, http->request->url.host()) != 0) {
+        /// XXX: fails on IP address-only hostname
```

Comment:
AFAICT, this matchDomainName() used to work for bracketed IPv6 addresses, with or without ports. matchDomainName() is not designed to compare IP addresses, but it appears to muddle through correctly in my basic tests (at least). In PR code, this check fails for all bracketed IPv6 addresses, right? Am I missing something? If my analysis is correct, then why would we want this kind of breaking change?!